aten/src/ATen/native/Unique.cpp

```diff
@@ -30,6 +30,74 @@ namespace native{
 
 namespace {
 
+// This unique implementation when dtype is bool is mapped
+// from UniqueCub.cu which uses a reduction to find the number of
+// true values.
+std::tuple<Tensor, Tensor, Tensor> unique_cpu_bool_template(
+    const Tensor& self,
+    const bool return_inverse,
+    const bool return_counts) {
+  const Tensor& input = self.contiguous();
+  bool* input_data = input.data_ptr<bool>();
```

Comment:
Does this work when the input is empty, or should we have a escape hatch in the function that calls into this one?

Reply:
Oh thanks for the remainder ! No this won't work properly when input is empty, it is fixed now!